fineract-provider/src/main/java/org/apache/fineract/portfolio/collateralmanagement/service/ClientCollateralManagementReadPlatformServiceImpl.java

```diff
@@ -88,6 +78,7 @@ public ClientCollateralManagementData getClientCollateralManagementData(final Lo
         for (LoanCollateralManagement loanCollateralManagement : loanCollateralManagementSet) {
             if (loanCollateralManagement.getLoanTransaction() != null) {
                 Long transactionId = loanCollateralManagement.getLoanTransaction().getId();
+                assert transactionId != null;
```

Comment:
Please don't use assert for validations.

Reply:
Understood, I've reverted the change. It wasn't required - I put an assertion because I saw an IDE warning. I've restored the code to what it was before